spec/integration/bulk_write_spec.rb

```diff
@@ -16,4 +16,35 @@
       end.not_to raise_error
     end
   end
+
+  context 'when bulk write needs to be split' do
+    let(:subscriber) { EventSubscriber.new }
+
+    let(:max_bson_size) { Mongo::Server::ConnectionBase::DEFAULT_MAX_BSON_OBJECT_SIZE }
+
+    let(:insert_events) do
+      subscriber.command_started_events('insert')
+    end
+
+    let(:failed_events) do
+      subscriber.failed_events
+    end
+
+    let(:operations) do
+      [{ insert_one: { text: 'a' * (max_bson_size/2) } }] * 3
```

Comment:
We appear to be splitting at server command size limit (48m rather than 16m) so bumping this to 6 would make the tests green.